src/main/kotlin/app/hashers/CommitCrawler.kt

```diff
@@ -156,6 +175,26 @@ object CommitCrawler {
                 catch (e: Exception) { null }
                 stream != null && !RawText.isBinary(stream)
             }
+            .filter { diff ->
+                val filePath =
+                    if (diff.getNewPath() != DiffEntry.DEV_NULL) {
+                        diff.getNewPath()
+                    } else {
+                        diff.getOldPath()
+                    }
+
+                // Update ignored paths list. The config file has retroactive
+                // force, i.e. if it was added at this commit, then we presume
+                // it is applied to all commits, preceding this commit.
+                if (diff.getOldPath() == CONF_FILE_PATH) {
+                    ignoredPaths =
+                        getIgnoredPaths(repo, diff.getOldId().toObjectId())
```

Comment:
Should be getNewId().

Reply:
If this will be newId(), then the config file content will be equal to the cached one, so we don't update its content when rolling back in history, until I miss anything.

Reply:
It's ok that it may read current config twice when processing newest commit. We should use newId. 
For example: two commits in repo. Oldest commit contains config, but it is empty. Newest commit adds ignore paths to config. Initially before crawling we will read current config with ignore path. When crawling newest commit we will take old content of file, which is empty. So config will not be taken into account. Also, with old we will not be able to read config from oldest commit.

Reply:
So you think the config changes should follow right after the change or within the change, but it cann't go before the change. So for example
-> conf: ++test1
-> +test1/
-> conf: --test1 ++ test2
-> -test1/ + test2
ignored list is test2, so test2 folder is ignored, then ignored list is test2 (newId), and thus test1 folder is processed

which is weird, because it sort of breaks natural order of commits reading flow, but might be ok. We just should point out that config is applied to all preceding commits until overridden.

so going with newId then?